### PR TITLE
Fix a crash when exiting the game with some mods

### PR DESCRIPTION
**Pull request by KingEnderBrine**
tl;dr; Since Unity vesrion update to 2019.4.37 some mods cause the game to crash when existing it, because of the way they create runtime prefabs.
Discord conversation: https://discord.com/channels/562704639141740588/562711077453037589/1257687310087225456

Some mods add prefabs loaded from asset bundles directly as children to runtime prefabs instantiated with PrefabsAPI.InstantiateClone. And it seems like these prefabs are destroyed before runtime prefabs when the game is closing, which results in a harmless, but certainly annoying crash.

Added null check to 2 native functions to avoid the crash. Added custom GenerateTrampoline method, because the one from MonoMod doesn't work with recursive functions.

### CODE REVIEW

**Review by xiaoxiao921 — APPROVED** [review]
Very clean, thanks!
